snuba/consumers/consumer_builder.py

```diff
@@ -172,15 +196,29 @@ def __init__(
         self.__commit_retry_policy = commit_retry_policy
 
     def __build_consumer(
-        self, strategy_factory: ProcessingStrategyFactory[KafkaPayload]
+        self,
+        strategy_factory: ProcessingStrategyFactory[KafkaPayload],
+        slice_id: Optional[int] = None,
     ) -> StreamProcessor[KafkaPayload]:
-        configuration = build_kafka_consumer_configuration(
+
+        # retrieves the default logical topic
+        topic = (
             self.storage.get_table_writer()
             .get_stream_loader()
             .get_default_topic_spec()
-            .topic,
+            .topic
+        )
+
+        # get the physical topic in case of slicing
+        if slice_id is not None:
+            physical_topic = SLICED_KAFKA_TOPIC_MAP[(topic.value, slice_id)]
+            topic = register_topic(physical_topic)
```

Comment:
You should only have to compute the `physical_topic` once throughout the consumer builder

Reply:
Are you suggesting making the physical topic (and therefore, also the logical topic) an instance attribute?

Reply:
I would say yes, unless I'm confused about how the consumer builder works and it builds many consumers per one instance of it

Reply:
In the new approach, `physical_topic` no longer exists in the constructor. So, I think this suggestion would no longer be possible/needed?